File: vscode_ros2_test_ws/src/ros2_cpp_pkg/src/subscriber.cpp
#include <iostream>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp" 


class HelloWorldSubNode : public rclcpp::Node
{
  public: 
    HelloWorldSubNode() : Node("hello_world_sub_node") 
		{
			subscription_ = this->create_subscription<std_msgs::msg::String>(
				"hello_world", 10, std::bind(&HelloWorldSubNode::sub_callback, this, std::placeholders::_1)
			);
		}

	private:
		void sub_callback(const std_msgs::msg::String & msg) const
		{
			RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg.data.c_str());
		}
		rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};


int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<HelloWorldSubNode>());
  rclcpp::shutdown();
  
  return 0;
}

File: vscode_ros2_test_ws/src/ros2_python_pkg/ros2_python_pkg/subscriber.py
#!/usr/bin/env python3
import rclpy   
from rclpy.node import Node    
from std_msgs.msg import String    

class HelloWorldSubscriber(Node):   
	def __init__(self):
		super().__init__("hello_world_sub_node")    
		self.sub = self.create_subscription(String, "hello_world", self.subscriber_callback, qos_profile=10)

	def subscriber_callback(self, msg):
		print("Recieved: " + msg.data)


def main(args=None):
	rclpy.init(args=args)     
	my_sub = HelloWorldSubscriber()    
	print("Waiting for data to be published...")

	try:
		rclpy.spin(my_sub)     
	except KeyboardInterrupt:   
		print("Terminating Node...")
		my_sub.destroy_node()      
		rclpy.shutdown()   

if __name__ == '__main__':
	main()